torch/testing/_internal/common_optimizers.py

```diff
@@ -104,12 +104,12 @@ class OptimizerInfo:
 
     def __init__(
         self,
-        optim_cls: Optimizer,  # Class object for the Optimizer under test
+        optim_cls: Type[Optimizer],  # Class object for the Optimizer under test
         *,
         # Function to generate optimizer inputs EXCLUDING params. We delegate params responsibility
         # to the test using the OptimizerInfo. OptimizerInput.params is likely None.
         # Can optionally take in device to filter out certain unsupported configs
-        optim_inputs_func,
+        optim_inputs_func: Callable[..., List[OptimizerInput]],
```

Comment:
Don't erase the args to the function, use a typing_extensions.ParamSpec variable.

Reply:
Ok! That sounds great

Reply:
They all seemed to have the same typing pattern so I used protocol, is that okay?